packages/talisman-ui/src/components/MysticalBackgroundV3/useCelestialArtifact.ts

```diff
@@ -1,95 +1,175 @@
 import Color from "color"
-import { SVGProps, useEffect, useMemo, useRef, useState } from "react"
+import { CSSProperties, useCallback, useLayoutEffect, useMemo, useRef, useState } from "react"
 
 import { MysticalPhysicsV3 } from "./MysticalPhysicsV3"
 
-export type ParentSize = {
-  width: number
-  height: number
-}
+export type CelestialArtifactProps = {
+  /** Configures the MysticalBackground parameters */
+  config: MysticalPhysicsV3
 
-export type ArtifactCharacteristics = {
-  ellipsis: SVGProps<SVGEllipseElement>
-  duration: number
-  color: string
-}
+  /** The size (in pixels) of the element we're rendering on */
+  parentSize: ParentSize
 
-const generateCharacteristics = (
-  config: MysticalPhysicsV3,
-  parentSize: ParentSize,
-  duration: number,
-  initialized: boolean,
-  forceColor?: string
-): ArtifactCharacteristics => {
-  const color = forceColor || Color.hsv(Math.random() * 360, 100, 100).hex()
-  const maxSize = Math.min(parentSize.width, parentSize.height)
-  const rx = Math.round(
-    maxSize * (config.radiusMin + Math.random() * (config.radiusMax - config.radiusMin))
-  )
-  const ry = Math.round(rx * (1 - config.ellipsisRatio * Math.random()))
-
-  const cx = Math.round(Math.random() * parentSize.width)
-  const cy = Math.round(Math.random() * parentSize.height - Math.max(rx, ry) / 2)
-
-  const ellipsis: SVGProps<SVGEllipseElement> = {
-    cx,
-    cy,
-    rx,
-    ry,
-    style: {
-      transitionProperty: "all",
-      transformBox: "fill-box",
-      transformOrigin: "center",
-      transform: `rotate(${Math.round(Math.random() * 360)}deg)`,
-      transitionDuration: `${duration}ms`,
-      transitionDelay: "100ms", // prevents flickering on FF
-      transitionTimingFunction: initialized ? "ease-in-out" : "ease-out",
-      opacity: Number(
-        (config.opacityMin + Math.random() * (config.opacityMax - config.opacityMin)).toFixed(2)
-      ),
-    },
-  }
+  /** Is used to ensure a consistent z-order of artifacts, i.e. to avoid any flickering from artifacts competing to be on top */
+  artifactIndex: number
```

Comment:
This manual z-order handling isn't *strictly* necessary, but I added it for good measure since we're doing 3d translations anyway (supposedly better performance because they're isolated from the rest of the page and sent to the GPU).